contracts/VerifierRegistry.sol

```diff
@@ -95,28 +106,34 @@ contract VerifierRegistry is Ownable {
     );
   }
 
-  function withdraw(uint256 _value) public returns (bool success) {
+  function withdraw(uint256 _value) public returns (bool) {
     Verifier storage verifier = verifiers[msg.sender];
 
-    require(_value > 0 && verifier.balance >= _value);
+    require(_value > 0 && verifier.balance >= _value, "nothing to withdraw");
 
     verifier.balance -= _value;
 
+    uint256 shard = verifier.shard;
+    uint256 shardBalance = totalBalancePerShard[shard] - _value;
+    totalBalancePerShard[shard] = shardBalance;
+
+    emit LogTotalBalancePerShard(shard, shardBalance);
```

Comment:
This should reflect the action/event that just happened. 
Something like: `LogVerifierWithdrawal` or `LogVerifierBalanceUpdate`

Reply:
Event for transfer/withdraw is present in `Token` contract.`LogTotalBalancePerShard` event is only about shardBalance.